src/sqlfluff/core/parser/segments/base.py

```diff
@@ -237,6 +238,7 @@ def __init__(
         segments,
         pos_marker: Optional[PositionMarker] = None,
         name: Optional[str] = None,
+        uuid: Optional[UUID] = None,
```

Comment:
I'm curious: Under what circumstances would a caller provide this parameter?

Reply:
It's effectively called during a `.copy()` operation - in theory you'd call it if you were ever making a new segment which you wanted to represent (or at least act as an anchor for) an existing segment. Going deeper into immutability might require that a little more.